### PR TITLE
Minimal validation of input XML

### PR DESCRIPTION
**Pull request by ralfhandl**
- check XML attributes: missing required attributes, unknown local attributes
- check XML structure: only allowed child elements with correct minimum and maximum occurrence

### CODE REVIEW

**Review by HeikoTheissen — COMMENTED** [review]
Unexpected attributes go undetected on elements that don't expect any. I have added code for the `Null` element, but there are others.

**Comment by ralfhandl** [comment]
@HeikoTheissen Good catch, will add similar code in all other cases.

File: test/xml2json.test.js [hunk 1]
```diff
@@ -519,7 +709,7 @@ describe("Error cases", function () {
     } catch (e) {
       assert.strictEqual(
         e.message.split("\n")[0],
-        "Element: Include, unexpected attribute: alias"
+        "Element: edmx:Include, unexpected attribute: alias"
```
Comment (HeikoTheissen) [inline test/xml2json.test.js:712]:
```suggestion
        "Element Include, unexpected attribute: alias"
```

File: lib/xml2json.js [hunk 2]
```diff
@@ -1394,30 +1755,18 @@ module.exports.xml2json = function (
...
-          ) &&
-          !(node.local === "Parameter" && a.local === "Mode") &&
-          !names.includes(a.local)
+          !extract.includes(a.local) &&
+          !ignore.includes(a.local)
       )
       .forEach((attribute) => {
         throw new Error(
-          `Element: ${node.local}, unexpected attribute: ${attribute.local}`
+          `Element: ${node.name}, unexpected attribute: ${attribute.name}`
```
Comment (HeikoTheissen) [inline lib/xml2json.js:1769]:
```suggestion
          `Element ${node.local}, unexpected attribute: ${attribute.name}`
```
Elsewhere, element names in error messages are not prefixed.

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]

**Review by HeikoTheissen — APPROVED** [review]

**Pull request merged**